src/tasks/component-installers/cert-manager.ts

```diff
@@ -0,0 +1,184 @@
+
+              // Await created resources to be available
+              await this.kubeHelper.waitServiceAccount(CA_CERT_GENERATION_SERVICE_ACCOUNT_NAME, CERT_MANAGER_NAMESPACE_NAME)
+
+              // Run CA key pair generation job
+              try {
+                await this.kubeHelper.createJob(CA_CERT_GENERATION_JOB_NAME, CA_CERT_GENERATION_JOB_IMAGE, CA_CERT_GENERATION_SERVICE_ACCOUNT_NAME, CERT_MANAGER_NAMESPACE_NAME)
+                await this.kubeHelper.waitJob(CA_CERT_GENERATION_JOB_NAME, CERT_MANAGER_NAMESPACE_NAME)
+              } catch {
+                command.error('Failed to generate self-signed CA certificate: generating job failed.')
+              }
+            } finally {
+              // Clean up resources
+              try {
+                // Do not change order of statements.
+                // Despite logically it is better to remove role binding first, we should delete items here in order of their creation.
+                // Such approach will resolve situation if only subset of items were created during previos run.
+                await this.kubeHelper.deleteServiceAccount(CA_CERT_GENERATION_SERVICE_ACCOUNT_NAME, CERT_MANAGER_NAMESPACE_NAME)
+                await this.kubeHelper.deleteRole('ca-cert-generator-role', CERT_MANAGER_NAMESPACE_NAME)
+                await this.kubeHelper.deleteRoleBinding('ca-cert-generator-role-binding', CERT_MANAGER_NAMESPACE_NAME)
+
+                await this.kubeHelper.deleteJob(CA_CERT_GENERATION_JOB_NAME, CERT_MANAGER_NAMESPACE_NAME)
+              } catch {
+                // Do nothing
+              }
+            }
+
+            // Wait until the secret is available
+            await this.kubeHelper.waitSecret('ca', CERT_MANAGER_NAMESPACE_NAME)
+          } else {
+            task.title = `${task.title}...already exists`
+          }
+        }
+      },
+      {
+        title: 'Set up Che certificates issuer',
+        task: async (_ctx: any, task: any) => {
+          const cheClusterIssuerExists = await this.kubeHelper.clusterIssuerExists('che-cluster-issuer')
+          if (!cheClusterIssuerExists) {
+            const cheCertificateClusterIssuerTemplatePath = path.join(flags.resources, '/cert-manager/che-cluster-issuer.yml')
+            await this.kubeHelper.createCheClusterIssuer(cheCertificateClusterIssuerTemplatePath)
+
+            task.title = `${task.title}...done`
+          } else {
+            task.title = `${task.title}...already exists`
+          }
+        }
+      },
+      {
+        title: 'Request self-signed certificate',
+        task: async (ctx: any, task: any) => {
+          if (ctx.cheCertificateExists) {
+            throw new Error('Che certificate already exists.')
+          }
+
+          const certificateTemplatePath = path.join(flags.resources, '/cert-manager/che-certificate.yml')
+          await this.kubeHelper.createCheClusterCertificate(certificateTemplatePath, flags.domain)
```

Comment:
I tried to deploy secure Che and reached timeout first time
![Screenshot_20200131_092437](https://user-images.githubusercontent.com/5887312/73520318-0573e880-440c-11ea-9ad7-07d0f27fbf2c.png)

I suppose it's because time is consumed by cert manager... I wonder if there is global timeout...

Then I tried to redo the same command and got the following
![Screenshot_20200131_092452](https://user-images.githubusercontent.com/5887312/73520301-fb51ea00-440b-11ea-83bc-7d81ff935453.png)


Please take a look issues I faced and consider improving UX here

Reply:
Thank you for letting me know. The problem is not in the timeout. I missed one place where namespace `che` was hardcoded. I've fixed that and now it can deploy Che in custom namespace as well.